src/sentry/api/endpoints/integrations/sentry_apps/rotate_secret.py

```diff
@@ -0,0 +1,62 @@
+from django.http import Http404
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import control_silo_endpoint
+from sentry.api.bases.sentryapps import SentryAppBaseEndpoint
+from sentry.api.permissions import SentryPermission
+from sentry.api.serializers import serialize
+from sentry.models.apiapplication import generate_token
+from sentry.models.integrations.sentry_app import SentryApp
+from sentry.services.hybrid_cloud.organization import organization_service
+from sentry.services.hybrid_cloud.user.service import user_service
+
+
+class SentryAppRotateSecretPermission(SentryPermission):
+    scope_map = {
+        "POST": ["org:write", "org:admin"],
+    }
+
+    def has_object_permission(self, request: Request, view: object, sentry_app: SentryApp):
+        # organization that owns an integration
+        org_context = organization_service.get_organization_by_id(id=sentry_app.owner_id)
```

Comment:
since this is an RPC call, which could have network/side effect errors, would it be worth wrapping in a try/catch, and defaulting to return false for safety?